agimus_controller_ros/reaching_goal_controller.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+import rospy
+import numpy as np
+from copy import deepcopy
+import time
+from threading import Lock
+from std_msgs.msg import Duration, Header
+from linear_feedback_controller_msgs.msg import Control, Sensor
+import atexit
+
+from agimus_controller.hpp_interface import HppInterface
+from agimus_controller_ros.ros_np_multiarray import to_multiarray_f64
+from agimus_controller.robot_model.panda_model import get_task_models
+from agimus_controller.utils.pin_utils import get_ee_pose_from_configuration
+from agimus_controller.mpc import MPC
+from agimus_controller.ocps.ocp_pose_ref import OCPPoseRef
+from agimus_controller_ros.sim_utils import convert_float_to_ros_duration_msg
+from agimus_controller_ros.parameters import AgimusControllerNodeParameters
+
+
+class ReachingGoalController:
```

Comment:
Why not inheriting from the base controller class?

Reply:
for this ros node we don't have a state machine, we don't have a hpp buffer, and because of that many functions are different so I didn't do inheriting, but maybe in the future we could make an even higher level base class which this one would inherate from ?